developers/ide/intellij.md

```diff
@@ -9,28 +9,42 @@ title: IntelliJ
+## Build the repositories
 
-1. Install the official [openHAB distribution](https://www.openhab.org/download/)
-1. Start the distribution **in debug mode** (use `./start_debug.sh` instead of `./start.sh` in step 4)
+1. Fork and clone the repositories into a parent directory (Reference `<PARENT_DIR>` from now on for this article). Take only the one(s) you will work on:
 
-This article refers to the directory where you installed the distribution as `<DISTRO_DIR>`.
+    - [openhab addons repository](https://www.github.com/openhab/openhab-addons)
+    - [openhab core repository](https://www.github.com/openhab/openhab-core)
+    - [openhab webui repository](https://www.github.com/openhab/openhab-webui)
 
-## Build the addons repository
+Use the command `git clone https://github.com/<yourgitusername>/openhab-<addons|core|webui>` (replace git user name accordingly).
 
-1. Fork and clone the [openhab addons repository](https://www.github.com/openhab/openhab-addons) into a new directory (Reference `<ADDON_DIR>` from now on for this article) with `git clone https://github.com/<yourgitusername>/openhab-addons` (replace git user name accordingly)
+1. Open IntelliJ, select the file/open, and choose the `<PARENT_DIR>`
+1. Open The Module settings (inside `Project Settings`, or use F4). Click on the + button and select `Import module`.
+1. Choose one of the repository directory you just cloned. select the Maven external model as import format.
+   IntelliJ will start importing and indexing. It will take while, wait until finished. The Module window should now be filled with a bunch of projects.
+   ![Import modules](images/ide_setup_intellij_import_module.png)
+1. Repeat step 2 and 3 for all repositories you cloned.
+1. Use Maven to clean & install projects
 
-1. Open IntelliJ and create a new project from existing sources (File | New | Project from existing sources) and pick `<ADDON_DIR>`/pom.xml
+    - `mvn clean install` in the root of the repositories using commandline Maven (or IntelliJ Maven view)
+    - some of the add-ons might fail to build - if it's not the one you're interested in, that should not bother you
+    - when the Maven project finished, you should find the freshly built addon JAR in the target directory
 
-    IntelliJ will start importing, indexing and building, it will take while, wait until finished (see status bar)
+You then have two main options to run your development. Use the official distribution, or use Maven.
 
-1. Use Maven to clean & install the addons project
+## Option 1: Use and debug from the official distribution
 
-    - mvn clean install in the root of `<ADDON_DIR>` using commandline Maven (or IntelliJ Maven view)
-    - some of the add-ons might fail to build - if it's not the one, you're interested in that should not bother you
-    - when the Maven project finished, you should find the freshly built addon JAR in the target directory
+This is the simpler option, but you can only use it for add-on development.
```

Comment:
Maybe we should link to the generic page here?

Reply:
I don't get it, which generic page ?
In the line you cite, the "simpler option" mentionned is described just after.

Reply:
developers/ide/generic.md

Reply:
The generic page is the option 2 section, and it already has a link.